src/components/RegistrationTable/EditCell.tsx

```diff
@@ -22,25 +22,29 @@ interface EditCellProps {
 }
 
 export const EditCell: React.FC<EditCellProps> = ({ row, table }) => {
-    const [questions, setQuestions] = useState<RegQuestionData[]>([]);
-    const [responses, setResponses] = useState<{ [key: string]: UserResponse }>({});
+    const [questions, setQuestions] = useState<RegistrationQuestion[]>([]);
 
     useEffect(() => {
-        if (router.isReady) {
-            const eventId = router.query.eventId as string;
-            const attendeeId = row.original.id;
-            const eventYear = router.query.year as string;
-
-            if (eventId && attendeeId && eventYear) {
-                fetchQuestionsAndResponses(attendeeId, eventId, eventYear).then(data => {
-                    setQuestions(data.questions);
-                    setResponses(data.responses);
-                }).catch(error => {
-                    console.error("Error fetching questions and responses:", error);
-                });
+        const fetchData = async () => {
+            if (router.isReady) {
+                const eventId = router.query.eventId as string;
+                const attendeeId = row.original.id;
+                const eventYear = router.query.year as string;
+    
+                if (eventId && attendeeId && eventYear) {
+                    try {
+                        const data = await fetchQuestionsAndResponses(attendeeId, eventId, eventYear);
+                        setQuestions(data.questions);
+                    } catch (error) {
+                        console.error("Error fetching questions and responses:", error);
+                    }
+                }
```

Comment:
This data will already be fetched by `src/pages/admin/event/[eventId]/[year]/index.tsx` and we can pass it to this Pop-up component instead of needing to re-fetch everything. Leave this in for now though, and we can refactor it once we have the backend working with the Admin Event Registrations page